xla/service/gpu/matmul_utils.cc

```diff
@@ -835,5 +835,43 @@ std::string TritonGemmConfig::ToString() const {
                       "}");
 }
 
+absl::StatusOr<bool> IsMatrixMultiplicationTooSmallForRewriting(
+    const HloInstruction& dot) {
+  CHECK_EQ(dot.opcode(), HloOpcode::kDot);
+
+  const Shape& lhs_shape = dot.operand(0)->shape();
+  const Shape& rhs_shape = dot.operand(1)->shape();
+  const DotDimensionNumbers& dot_dims = dot.dot_dimension_numbers();
+
+  absl::Span<const int64_t> lhs_contracting_dims =
+      dot_dims.lhs_contracting_dimensions();
+  const int64_t contracting_size = absl::c_accumulate(
+      lhs_contracting_dims, 1, [&](int64_t size, int64_t dim) {
+        return size * lhs_shape.dimensions(dim);
+      });
+
+  TF_ASSIGN_OR_RETURN(
+      std::vector<int64_t> lhs_non_contracting_dims,
+      GetNonContractingDims(lhs_shape, dot_dims.lhs_batch_dimensions(),
+                            dot_dims.lhs_contracting_dimensions()));
+  const int64_t lhs_non_contracting_size = absl::c_accumulate(
+      lhs_non_contracting_dims, 1, [&](int64_t size, int64_t dim) {
+        return size * lhs_shape.dimensions(dim);
+      });
+
+  TF_ASSIGN_OR_RETURN(
+      std::vector<int64_t> rhs_non_contracting_dims,
+      GetNonContractingDims(rhs_shape, dot_dims.rhs_batch_dimensions(),
+                            dot_dims.lhs_contracting_dimensions()));
+  const int64_t rhs_non_contracting_size = absl::c_accumulate(
+      rhs_non_contracting_dims, 1, [&](int64_t size, int64_t dim) {
+        return size * lhs_shape.dimensions(dim);
```

Comment:
lhs_shape -> rhs_shape?

Reply:
Shouldn't this be:
absl::c_accumulate(rhs_non_contracting_dims, 1, std::multiplies<int64_t>());
instead?

Reply:
> lhs_shape -> rhs_shape?

Good catch!

> Shouldn't this be:
> absl::c_accumulate(rhs_non_contracting_dims, 1, std::multiplies<int64_t>());
> instead?

I think it's correct as is, `rhs_non_contracting_dims` really contains dimension indexes, not sizes. The shape has the sizes.